### PR TITLE
HighNodeUtilisation strategy

### LINKED ISSUE #196: Add HighNodeUtilization strategy
I want to keep my cluster as compact as possible. In order to do that, I switched kubernetes-scheduler policy from LeastRequestedPriority to MostRequestedPriority. It would be greate to have corresponding HighNodeUtilization strategy for descheduler.

### PR DESCRIPTION
**Pull request by hanumanthan**
Adding `HighNodeutilisation` strategy. Accepts threshold and targetThreshold as parameter configs.
Eliminate pods from nodes under targetThreshold and places them in nodes between threshold and targetThreshold.
This works in tandem to scheduler config `MostRequestedPriority` such that pods are placed in nodes with high utilization.

Functions common to both High and LowNode Utilization strategies have been extracted to a separate file.

Fixes #196 

This is a new PR to fix comments from [here](https://github.com/kubernetes-sigs/descheduler/pull/339)

### CODE REVIEW

**Comment by damemi** [comment]
@hanumanthan is this PR still a draft or ready for for review? Just curious because I saw you tagged me, but I don't know if it's ready for feedback or if you're still working on it

**Comment by hanumanthan** [comment]
> @hanumanthan is this PR still a draft or ready for for review? Just curious because I saw you tagged me, but I don't know if it's ready for feedback or if you're still working on it

Hi @damemi, The PR is draft because these changes are pending. The main strategy changes are done and want to get feedback on the approach before I work on these

- E2E tests
- Docs update
- Config updates

**Comment by hanumanthan** [comment]
@damemi @ingvagabund Can you please review this. Thanks

File: pkg/descheduler/strategies/highnodeutilization.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		klog.V(1).Infof("evicting pods based on priority, if they have same priority, they'll be evicted based on QoS tiers")
+		// sort the evictable Pods based on priority. This also sorts them based on QoS. If there are multiple pods with same priority, they are sorted based on QoS tiers.
+		podutil.SortPodsBasedOnPriorityLowToHigh(removablePods)
+		evictPodsFromHighUsageNode(ctx, removablePods, &totalPods, &totalCPU, &totalMem, taintsOfDesiredNodes, podEvictor, node.node)
+
+		klog.V(1).Infof("%v pods evicted from node %#v with usage %v", podEvictor.NodeEvicted(node.node), node.node.Name, node.usage)
+	}
+}
+
+func evictPodsFromHighUsageNode(
```
Comment (ingvagabund) [inline pkg/descheduler/strategies/highnodeutilization.go:150]: `evictPods` and `evictPodsFromHighUsageNode` should be quite similar. Or, identical. At the end it's just about selecting a list of nodes from which a set of pods get evicted while still checking sufficient amount of resources is still available.

**Comment by ingvagabund** [comment]
@hanumanthan https://github.com/kubernetes-sigs/descheduler/pull/384 refactors the code to provide more reusable bits

**Comment by seanmalloy** [comment]
Greetings @hanumanthan we just completed the descheduler v0.19.0(k8s v1.19) release cycle. We are starting to work on the features for the descheduler v0.20.0(k8s v1.20.0) release. Are you planning on continuing to work on this feature enhancement?

It looks like this work will be easier to complete once #384 is merged(hopefully soon).

**Comment by hanumanthan** [comment]
Closing this PR as its outdated with the lownodeutilization refractor. Will open a fresh PR keeping this comments in mind

**Pull request closed without merging**